### ISSUE TITLE
Skin ext_config overwriting wrong parts of main ext_config

### ISSUE THREAD

**Issue by Trava715** [ac-custom-shaders-patch/acc-extension-config#169]
When using skin folder ext_config to add per-skin adjustments, any changes overwrite the sections of main ext_config instead of being added at the end.

 What happpens currently: 
![image](https://user-images.githubusercontent.com/43389722/149296345-139d96b9-a1ab-4214-9922-b0618ebc95ef.png)

What should happen: 
![image](https://user-images.githubusercontent.com/43389722/149296397-eaba3d9d-ff0c-4eed-9da5-738fadaabb69.png)

This causes a lot of unnecessary clutter in config by having to add empty headers as "free slots" for any skin adjustments like this:
![image](https://user-images.githubusercontent.com/43389722/149296654-b8894053-6518-4769-9fa0-6bff96ae30c0.png)

While it's not a big issue with carpaint materials it becomes a huge mess when trying to do a PBR replacement since you need to copy the entire PBR section to skin config just to change one material.

**Comment by Marocco2** [comment]
This is the intended usage, isn't it? Car paint should be defined in ext_config car and then change relevant params per skin

**Comment by Trava715** [comment]
> Car paint should be defined in ext_config car and then change relevant params per skin

Yes but if you have any paint that should be applied to all skins globally eg. calipers and put it in main config, then skin carpaint overwrites it. Hence having to add empty headers.

Same with PBR, let me post example

```
;PBR INTERIOR
[Material_Velvet]
Materials = INT_ALCANTARA
DetailScale = 20
ApplyTilingFix=1
OcclusionMult = 1.35
RESOURCE_0 = txDetail
RESOURCE_FILE_0 = ALCANTARA.dds
RESOURCE_1 = txNormalDetail
RESOURCE_FILE_1 = ALCANTARA_NM.dds

[Material_Leather_v2]
Materials = INT_LEATHER
ApplyTilingFix=1
DetailScale=75
Smoothness=0.2 
Brightness=0.35

[Material_Carbon]
Meshes = INT_PLASTIC_TRIM
Brightness = 2
ClearCoatF0 = 0.35
DetailScale = 1000
CubemapReflectionBlur = 0.0
```

If that's my  EXT_CONFIG code and I want to change the INT_LEATHER to be something else per skin then I need to copy **the entire config leading up to it** or else it'll overwrite the first PBR entry - in this case changing INT_LEATHER per skin would replace INT_ALCANTARA.

**Comment by Trava715** [comment]
@Marocco2 

https://user-images.githubusercontent.com/43389722/149400346-a3f6e038-3f63-49ba-bb98-f20e94bd67ed.mp4

**Comment by Marocco2** [comment]
> @Marocco2
> 
>  2022-01-13.19-55-42.mp4 
This applies even when you restart AC? (partial reload would be ignored)

You may find interesting using SKIN variable inside ext_config.ini so you can technically reuse the already PBR config without writing skin config at all. (speculation)

> > What if you set materials in ext_config as TEMPLATE? That would solve partially your issues
> 
> Elaborate? Like use empty headers of `[Material_xxxxxxx]` same as I'm doing with carpaint headers?

I may have a clear image of the issue now.
Take this as a pinch of salt but IIRC there may be a switch for skin config that ignores everything inside ext_config (described somewhere in a CSP changelog probably)
I should check if it is real or not.

**Comment by Marocco2** [comment]
Or perhaps according to INIpp documentation there are some race conditions involved here when using templates (AKA PBR stuff)
If the ordering is skin config > ext_config > repo config (ignored if there is ext_config) the first `_...` shaders are taken by skin config and for some reason the ordering is reset when CSP is now looking to ext_config. I would suggest include in skin config in INCLUDE section this line:
`[INCLUDE: ext_config.ini]`

and then add leave your skin config untouched

**Comment by Trava715** [comment]
> there may be a switch for skin config that ignores everything inside ext_config

Only one that i'm aware of is IGNORE_OTHER_CONFIGS but I'm not using that.

**Comment by gro-ove** [comment]
You can also set section name explicitly when using a template, like so:
```
[SHADER_REPLACEMENT_0_MYPAINT : Material_CarPaint]
```

(Another thing is that when CSP collects sections like “SHADER_REPLACEMENT”, it picks all sections which start with “SHADER_REPLACEMENT_” followed by a digit and not end with “_”. You can use it to set explicit names to different sections, like “SHADER_REPLACEMENT_0_MYPAINT”, guaranteeing there won’t be a collision while still making it easy to override value in subsequent configs if needed.)

**Comment by Marocco2** [comment]
Still present in 0.1.78?
